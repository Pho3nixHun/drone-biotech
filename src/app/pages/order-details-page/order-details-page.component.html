@if (vm(); as vm) {
    <app-page-layout *transloco="let t">
        <app-header>
            <h2>
                {{
                    t(vm.headerXVM.idTitleKey, {
                        id: vm.headerXVM.id,
                    })
                }}
            </h2>

            <app-status [vm]="vm.headerXVM.statusXVM">
                {{ t(vm.headerXVM.statusXVM.statusTextKey) }}
            </app-status>

            <a [routerLink]="vm.headerXVM.addNewMissionsButtonXVM.routerLink">
                <app-button [vm]="vm.headerXVM.addNewMissionsButtonXVM">
                    @if (vm.headerXVM.addNewMissionsButtonXVM.textKey) {
                        {{ t(vm.headerXVM.addNewMissionsButtonXVM.textKey) }}
                    }
                </app-button>
            </a>

            <app-summary-list>
                @for (
                    summary of vm.headerXVM.summaryListXVM.summaryXVMs;
                    track $index
                ) {
                    <app-summary>
                        <p class="text-gray-600">
                            {{ t(summary.textKey) }}
                        </p>

                        <p class="font-medium">
                            @if (summary.value.type === 'withValueKey') {
                                {{ t(summary.value.key, summary.value.params) }}
                            } @else {
                                {{ summary.value.value }}
                            }
                        </p>
                    </app-summary>
                }
            </app-summary-list>
        </app-header>

        @for (sectionCardXVM of vm.sectionCardXVMs; track $index) {
            @switch (sectionCardXVM.type) {
                @case ('orderDetails') {
                    <app-section-card [vm]="sectionCardXVM" *transloco="let t">
                        <div
                            class="flex lg:flex-row flex-col space-y-8 lg:space-y-0"
                        >
                            @for (
                                infoPanelXVM of sectionCardXVM.infoPanelXVMs;
                                track $index
                            ) {
                                <app-info-panel class="lg:w-1/2 w-full">
                                    <h4 class="font-medium">
                                        {{ t(infoPanelXVM.titleKey) }}
                                    </h4>
                                    <app-info-item-list>
                                        @for (
                                            infoItemXVM of infoPanelXVM
                                                .infoListXVM.infoItemXVMs;
                                            track $index
                                        ) {
                                            <app-info-item>
                                                <dt class="text-gray-600 w-32">
                                                    {{
                                                        t(infoItemXVM.labelKey)
                                                    }}
                                                </dt>
                                                <dd>
                                                    <strong class="font-medium">
                                                        @if (
                                                            infoItemXVM.value
                                                                .type ===
                                                            'withValueKey'
                                                        ) {
                                                            {{
                                                                t(
                                                                    infoItemXVM
                                                                        .value
                                                                        .key,
                                                                    infoItemXVM
                                                                        .value
                                                                        .params
                                                                )
                                                            }}
                                                        } @else {
                                                            {{
                                                                infoItemXVM
                                                                    .value.value
                                                            }}
                                                        }
                                                    </strong>
                                                </dd>
                                            </app-info-item>
                                        }
                                    </app-info-item-list>
                                </app-info-panel>
                            }
                        </div>
                    </app-section-card>
                }
                @case ('orderActions') {
                    <app-section-card [vm]="sectionCardXVM" *transloco="let t">
                        <div class="join">
                            <a
                                [href]="
                                    sectionCardXVM.completionTemplateButtonXVM
                                        .link.href
                                "
                                download
                            >
                                <app-button
                                    [vm]="
                                        sectionCardXVM.completionTemplateButtonXVM
                                    "
                                >
                                    @if (
                                        sectionCardXVM
                                            .completionTemplateButtonXVM.textKey
                                    ) {
                                        {{
                                            t(
                                                sectionCardXVM
                                                    .completionTemplateButtonXVM
                                                    .textKey
                                            )
                                        }}
                                    }
                                </app-button>
                            </a>

                            <app-button
                                [vm]="sectionCardXVM.closeOrderButtonXVM"
                                (click)="confirmationDialog.openDialog()"
                            >
                                @if (
                                    sectionCardXVM.closeOrderButtonXVM.textKey
                                ) {
                                    {{
                                        t(
                                            sectionCardXVM.closeOrderButtonXVM
                                                .textKey
                                        )
                                    }}
                                }
                            </app-button>
                        </div>
                    </app-section-card>
                }
                @case ('messages') {
                    <app-section-card [vm]="sectionCardXVM" *transloco="let t">
                        <app-message-item-list>
                            @for (
                                messageItemXVM of sectionCardXVM
                                    .messageItemListXVM.messageItemXVMs;
                                track $index
                            ) {
                                <app-message-item>
                                    @switch (messageItemXVM.avatarXVM.type) {
                                        @case ('withInitials') {
                                            <app-avatar
                                                [class]="
                                                    messageItemXVM.avatarXVM
                                                        .colors
                                                "
                                            >
                                                {{
                                                    messageItemXVM.avatarXVM
                                                        .initials
                                                }}
                                            </app-avatar>
                                        }

                                        @case ('withImage') {
                                            <app-avatar
                                                class="text-white *:bg-black"
                                            >
                                                <img
                                                    [ngSrc]="
                                                        messageItemXVM.avatarXVM
                                                            .imageSrc
                                                    "
                                                    [alt]="
                                                        messageItemXVM.avatarXVM
                                                            .altTextKey
                                                    "
                                                    class="object-contain rounded-full"
                                                    fill
                                                />
                                            </app-avatar>
                                        }
                                    }
                                    <h3 class="font-medium">
                                        {{
                                            t(messageItemXVM.senderValueKey, {
                                                role: t(
                                                    messageItemXVM.roleTextKey
                                                ),
                                                name: messageItemXVM.senderName,
                                            })
                                        }}
                                    </h3>
                                    <p class="text-gray-700">
                                        {{ messageItemXVM.message }}
                                    </p>
                                    <time class="text-gray-600 text-sm">{{
                                        t(messageItemXVM.sendingDateValueKey, {
                                            date: messageItemXVM.sendingDate,
                                        })
                                    }}</time>
                                </app-message-item>
                            }
                        </app-message-item-list>
                        <div class="w-full flex flex-row">
                            <input
                                type="text"
                                [formControl]="messageFormControl"
                                class="px-2 w-full border border-gray-300"
                            />
                            <app-button
                                [vm]="sectionCardXVM.submitButtonXVM"
                                [disabled]="messageFormControl.invalid"
                                (click)="onSendMessageClick()"
                            >
                                @if (sectionCardXVM.submitButtonXVM.icon) {
                                    <mat-icon>{{
                                        t(sectionCardXVM.submitButtonXVM.icon)
                                    }}</mat-icon>
                                }
                            </app-button>
                        </div>
                    </app-section-card>
                }
            }
        }
    </app-page-layout>

    <app-confirmation-dialog
        #confirmationDialog
        [vm]="vm.confirmationDialogVM"
        (response)="onConfirmationDialogResponse($event)"
    />
}
