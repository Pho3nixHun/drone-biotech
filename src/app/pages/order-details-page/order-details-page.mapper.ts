import { mapAvatarXVM } from '@components/avatar/avatar.model';
import {
    HeaderXVM,
    MessageItemXVM,
    MessagesSectionCardXVM,
    OrderActionsSectionCardXVM,
    OrderDetailsPageConfig,
    OrderDetailsSectionCardXVM,
    OrderStatus,
    OverviewSectionCardXVM,
    Role,
    TargetArea,
} from './order-details-page.model';
import { Message, Order } from '@services/order/order.service.model';
import { isActiveMission } from '@services/mission/mission.service.model';

export const mapOrderStatusToStatusToCSSStyles = (
    status: OrderStatus
): string =>
    ({
        active: '*:bg-green-200 *:text-green-800',
        pending: '*:bg-red-200 *:text-red-800',
        completed: '*:bg-orange-200 *:text-orange-800',
    })[status] ?? '';

export const mapOrderStatusToTranslocoTextKey = (status: OrderStatus): string =>
    ({
        active: 'OrderDetailsPage.header.status.active',
        pending: 'OrderDetailsPage.header.status.pending',
        completed: 'OrderDetailsPage.header.status.completed',
    })[status] ?? '';

export const mapRoleTranslocoTextKey = (role: Role): string =>
    ({
        customer: 'OrderDetailsPage.role.customer',
        office: 'OrderDetailsPage.role.office',
        pilot: 'OrderDetailsPage.role.pilot',
    })[role] ?? '';

export const mapHeaderXVM = (
    config: OrderDetailsPageConfig,
    order: Order
): HeaderXVM => ({
    ...config.headerConfig,
    addMissionButtonXVM: {
        isVisible: order.status !== 'completed',
        textKey: config.headerConfig.addMissionButtonXVM.textKey,
    },
    id: order.id,
    statusXVM: {
        styles: mapOrderStatusToStatusToCSSStyles(order.status),
        statusTextKey: mapOrderStatusToTranslocoTextKey(order.status),
    },
    summaryListXVM: {
        summaryXVMs: [
            {
                textKey: config.headerConfig.clientTextKey,
                value: { type: 'withoutValueKey', value: order.client.client },
            },
            {
                textKey: config.headerConfig.createdDateTextKey,
                value: {
                    type: 'withValueKey',

                    key: config.headerConfig.createdDateValueKey,
                    params: { date: order.creationDate },
                },
            },
            {
                textKey: config.headerConfig.totalAreaTextKey,
                value: {
                    type: 'withValueKey',
                    key: config.headerConfig.totalAreaValueKey,
                    params: { area: order.totalAreaInHa },
                },
            },
        ],
    },
});

export const mapOrderActionsSectionCardXVM = (
    config: OrderDetailsPageConfig,
    order: Order
): OrderActionsSectionCardXVM => ({
    ...config.sectionCardConfigs.orderActionsSectionCardConfig,
    closeOrderButtonXVM: {
        ...config.sectionCardConfigs.orderActionsSectionCardConfig
            .closeOrderButtonXVM,
        isDisabled: order.status === 'completed',
    },
});

export const mapOrderDetailsSectionCardXVM = (
    config: OrderDetailsPageConfig,
    order: Order
): OrderDetailsSectionCardXVM => ({
    ...config.sectionCardConfigs.orderDetailsSectionCardConfig,
    infoPanelXVMs: [
        {
            ...config.sectionCardConfigs.orderDetailsSectionCardConfig
                .infoPanelConfig.clientInfoPanel,
            infoListXVM: {
                infoItemXVMs: [
                    {
                        labelKey:
                            config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .clientInfoPanel.contactLabelKey,
                        value: {
                            type: 'withoutValueKey',
                            value: order.client.contact,
                        },
                    },
                    {
                        labelKey:
                            config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .clientInfoPanel.emailLabelKey,
                        value: {
                            type: 'withoutValueKey',
                            value: order.client.email,
                        },
                    },
                    {
                        labelKey:
                            config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .clientInfoPanel.phoneLabelKey,
                        value: {
                            type: 'withoutValueKey',
                            value: order.client.phone,
                        },
                    },
                    {
                        labelKey:
                            config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .clientInfoPanel.addressLabelKey,

                        value: {
                            type: 'withoutValueKey',
                            value: order.client.address,
                        },
                    },
                ],
            },
        },
        {
            ...config.sectionCardConfigs.orderDetailsSectionCardConfig
                .infoPanelConfig.summaryInfoPanel,
            infoListXVM: {
                infoItemXVMs: [
                    {
                        labelKey:
                            config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .summaryInfoPanel.treatmentLabelKey,
                        value: {
                            type: 'withoutValueKey',
                            value: order.summary.treatment,
                        },
                    },
                    {
                        labelKey:
                            config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .summaryInfoPanel.averageDoseLabelKey,
                        value: {
                            type: 'withValueKey',
                            key: config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .summaryInfoPanel.averageDoseValueKey,
                            params: {
                                dose: order.summary.averageDose,
                            },
                        },
                    },
                    {
                        labelKey:
                            config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .summaryInfoPanel.totalSupplyLabelKey,
                        value: {
                            type: 'withValueKey',
                            key: config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .summaryInfoPanel.totalSupplyValueKey,
                            params: {
                                amount: order.summary.totalSupply,
                            },
                        },
                    },
                    {
                        labelKey:
                            config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .summaryInfoPanel.orderValueLabelKey,
                        value: {
                            type: 'withValueKey',
                            key: config.sectionCardConfigs
                                .orderDetailsSectionCardConfig.infoPanelConfig
                                .summaryInfoPanel.orderValueValueKey,
                            params: {
                                price: order.summary.orderValue,
                            },
                        },
                    },
                ],
            },
        },
    ],
});

export const mapMessagesSectionCardXVM = (
    config: OrderDetailsPageConfig,
    messages: Message[]
): MessagesSectionCardXVM => ({
    type: 'orderMessages',
    buttonTextKey:
        config.sectionCardConfigs.messagesSectionCardConfig.buttonTextKey,
    titleKey: config.sectionCardConfigs.messagesSectionCardConfig.titleKey,
    messageItemListXVM: {
        messageItemXVMs: messages
            .map((message) => mapMessageToMessageItemXVM(config, message))
            .sort(
                (a, b) =>
                    b.sendingDate.getSeconds() - a.sendingDate.getSeconds()
            ),
    },
});

const mapMessageToMessageItemXVM = (
    config: OrderDetailsPageConfig,
    message: Message
): MessageItemXVM => ({
    ...message,
    avatarXVM: mapAvatarXVM(message.sender),
    roleTextKey: mapRoleTranslocoTextKey(message.sender.role),
    senderName: message.sender.name,
    senderValueKey:
        config.sectionCardConfigs.messagesSectionCardConfig.senderValueKey,
    sendingDateValueKey:
        config.sectionCardConfigs.messagesSectionCardConfig.dateValueKey,
});

export const mapMissionsToTargetArea = (order: Order): TargetArea[] =>
    order.missions.map((missions) => ({
        coordinates: missions.coordinates,
        type: isActiveMission(missions) ? 'active' : 'completed',
    }));

export const mapOrderOverviewSectionCardXVM = (
    config: OrderDetailsPageConfig,
    order: Order
): OverviewSectionCardXVM => ({
    type: 'overview',
    titleKey: config.sectionCardConfigs.orderOverviewSectionCardConfig.titleKey,
    mapFormControlXVM: {
        targetAreas: mapMissionsToTargetArea(order),
        totalMissionsLabelKey:
            config.sectionCardConfigs.orderOverviewSectionCardConfig
                .totalMissionsLabelKey,
        totalMissionsValueKey:
            config.sectionCardConfigs.orderOverviewSectionCardConfig
                .totalMissionsValueKey,
        completedMissionsLabelKey:
            config.sectionCardConfigs.orderOverviewSectionCardConfig
                .completedMissionsLabelKey,
        completedMissionsValueKey:
            config.sectionCardConfigs.orderOverviewSectionCardConfig
                .completedMissionsValueKey,
        remainingMissionsLabelKey:
            config.sectionCardConfigs.orderOverviewSectionCardConfig
                .remainingMissionsLabelKey,
        remainingMissionsValueKey:
            config.sectionCardConfigs.orderOverviewSectionCardConfig
                .remainingMissionsValueKey,
    },
});
