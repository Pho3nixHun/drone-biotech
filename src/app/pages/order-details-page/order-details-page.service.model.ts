export type Role = 'customer' | 'office' | 'pilot';

export interface Sender {
    role: Role;
    name: string;
    photoUrl: string | null;
}

export interface Message {
    sender: Sender;
    sendingDate: Date;
    message: string;
}
