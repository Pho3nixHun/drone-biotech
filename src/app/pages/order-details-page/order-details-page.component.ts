import {
    ConfirmationDialogReason,
    ConfirmationDialogVM,
} from '@components/confirmation-dialog/confirmation-dialog.model';
import { FormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
import { toSignal } from '@angular/core/rxjs-interop';
import { Component, inject, signal } from '@angular/core';
import { NgClass, NgOptimizedImage } from '@angular/common';
import { TranslocoModule } from '@jsverse/transloco';
import { firstValueFrom } from 'rxjs';
import { Store } from '@ngrx/store';
import { selectUser } from '@stores/auth/auth.selector';
import { DialogService } from '@services/dialog/dialog.service';
import { SummaryListComponent } from '@components/summary-list/summary-list.component';
import { StatusComponent } from '@components/status/status.component';
import { SummaryComponent } from '@components/summary-list/components/summary/summary.component';
import { HeaderComponent } from './components/header/header.component';
import { PageLayoutComponent } from '@components/page-layout/page-layout.component';
import { SectionCardComponent } from '@components/section-card/section-card.component';
import { InfoPanelComponent } from '@components/info-panel/info-panel.component';
import { InfoItemListComponent } from '@components/info-panel/components/info-item-list/info-item-list.component';
import { AvatarComponent } from '@components/avatar/avatar.component';
import { InfoItemComponent } from '@components/info-panel/components/info-item-list/components/info-item/info-item.component';
import { ConfirmationDialogComponent } from '@components/confirmation-dialog/confirmation-dialog.component';
import { MessageItemListComponent } from '@components/message-item-list/message-item-list.component';
import { MessageItemComponent } from '@components/message-item/message-item.component';
import { emptyStringValidator } from '@utils/empty-string.validator';
import { OrderDetailsPageService } from './order-details-page.service';
import { GoogleMapComponent } from '@components/google-map/google-map.component';
import { DrawTargetAreasDirective } from '@components/google-map/directives/draw-target-areas/draw-target-areas.directive';
import { CardGroupComponent } from '@components/card-group/card-group.component';
import { CardGroupHeaderComponent } from '@components/card-group/components/card-group-header/card-group-header.component';
import { CardItemListComponent } from '@components/card-group/components/card-item-list/card-item-list.component';
import { CardItemComponent } from '@components/card-group/components/card-item-list/components/card-item/card-item.component';
import { CardItemContentComponent } from '@components/card-group/components/card-item-list/components/card-item/components/card-item-content/card-item-content.component';
import { KeyValueComponent } from '@components/key-value/key-value.component';
import { CardItemActionListComponent } from '@components/card-group/components/card-item-list/components/card-item/components/card-item-action-list/card-item-action-list.component';
import { RouterModule } from '@angular/router';
import { MatIconModule } from '@angular/material/icon';
import { ValueComponent } from '@components/value/value.component';
import { Status } from './order-details-page.model';
import { Avatar } from '@components/avatar/avatar.model';

@Component({
    selector: 'app-order-details-page',
    imports: [
        PageLayoutComponent,
        HeaderComponent,
        TranslocoModule,
        MessageItemListComponent,
        MessageItemComponent,
        StatusComponent,
        SummaryListComponent,
        SummaryComponent,
        SectionCardComponent,
        InfoPanelComponent,
        InfoItemListComponent,
        InfoItemComponent,
        ReactiveFormsModule,
        NgClass,
        AvatarComponent,
        NgOptimizedImage,
        GoogleMapComponent,
        DrawTargetAreasDirective,
        ValueComponent,
        CardGroupComponent,
        CardGroupHeaderComponent,
        CardItemListComponent,
        CardItemComponent,
        CardItemContentComponent,
        KeyValueComponent,
        CardItemActionListComponent,
        RouterModule,
        MatIconModule,
    ],
    templateUrl: './order-details-page.component.html',
})
export class OrderDetailsPageComponent {
    private readonly store = inject(Store);
    private readonly fb = inject(FormBuilder);
    private readonly dialogService = inject(DialogService);
    private readonly orderDetailsPageService = inject(OrderDetailsPageService);

    protected readonly vm = toSignal(this.orderDetailsPageService.getVM());

    protected readonly drawnPolygons = signal<google.maps.Polygon[] | null>(
        null
    );

    protected readonly messageFormControl = this.fb.control('', [
        Validators.required,
        emptyStringValidator(),
    ]);

    protected async onSendMessageClick() {
        const message = this.messageFormControl.value;

        if (this.messageFormControl.invalid || !message) return;

        const sender = await firstValueFrom(this.store.select(selectUser));

        if (!sender) return;

        this.orderDetailsPageService.sendMessage({
            sendingDate: new Date(),
            message,
            sender: {
                role: sender.role,
                name: sender.displayName,
                photoUrl: sender.photoURL,
            },
        });

        this.messageFormControl.reset();
    }

    protected async openConfirmationDialog(vm: ConfirmationDialogVM) {
        const reason: ConfirmationDialogReason = await firstValueFrom(
            this.dialogService.create(vm, ConfirmationDialogComponent).result$
        );

        if (reason.reasonType === 'cancel') return;
        return this.orderDetailsPageService.closeOrder();
    }

    protected readonly Status = Status;
    protected readonly Avatar = Avatar;
}
