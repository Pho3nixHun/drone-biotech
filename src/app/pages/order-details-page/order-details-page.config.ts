import { InjectionToken } from '@angular/core';
import { orderDetailsPageConfig as config } from './order-details-page.mock';
import { OrderDetailsPageConfig } from './order-details-page.model';
import {
    ACTIVE_MISSION_POLYGON_OPTIONS,
    COMPLETED_MISSION_POLYGON_OPTIONS,
} from 'src/app/shared/providers/google-maps-provider';
import { ConfigProvider } from '@interfaces/config-provider';

export const ORDER_DETAILS_PAGE_CONFIG =
    new InjectionToken<OrderDetailsPageConfig>('Order details page config');

export const orderDetailsPageConfig: ConfigProvider[] = [
    {
        provide: ORDER_DETAILS_PAGE_CONFIG,
        useValue: config,
    },
    {
        provide: ACTIVE_MISSION_POLYGON_OPTIONS,
        useValue: {
            clickable: false,
            draggable: false,
            editable: false,
            fillOpacity: 0.5,
            strokeColor: 'green',
            fillColor: 'green',
        },
    },
    {
        provide: COMPLETED_MISSION_POLYGON_OPTIONS,
        useValue: {
            clickable: false,
            draggable: false,
            editable: false,
            fillOpacity: 0.5,
            strokeColor: 'orange',
            fillColor: 'orange',
        },
    },
];
