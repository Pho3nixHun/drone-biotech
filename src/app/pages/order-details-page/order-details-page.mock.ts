import { MatIcon } from '@interfaces/mat-icon.enum';
import { OrderDetailsPageConfig } from './order-details-page.model';

export const orderDetailsPageConfig: OrderDetailsPageConfig = {
    confirmationDialogVM: {
        closeButtonXVM: {
            icon: MatIcon.CLOSE,
            variant: 'ghost',
            secondary: true,
        },
        titleKey: 'OrderDetailsPage.orderActions.close.dialog.title',
        confirmTextKey:
            'OrderDetailsPage.orderActions.close.dialog.confirmText',
        cancelButtonXVM: {
            textKey:
                'OrderDetailsPage.orderActions.close.dialog.cancelButtonText',
            secondary: true,
            variant: 'fill',
        },
        confirmButtonXVM: {
            textKey:
                'OrderDetailsPage.orderActions.close.dialog.confirmButtonText',
            variant: 'fill',
        },
    },
    headerConfig: {
        idTitleKey: 'OrderDetailsPage.header.idTitle',
        clientTextKey: 'OrderDetailsPage.header.clientText',
        createdDateTextKey: 'OrderDetailsPage.header.createdDateText',
        createdDateValueKey: 'OrderDetailsPage.header.createdDateValue',
        totalAreaTextKey: 'OrderDetailsPage.header.totalAreaText',
        totalAreaValueKey: 'OrderDetailsPage.header.totalAreaValue',
        addMissionButtonXVM: {
            secondary: false,
            variant: 'ghost',
            textKey: 'OrderDetailsPage.header.addNewMissionButtonText',
        },
    },
    sectionCardConfigs: {
        messagesSectionCardConfig: {
            buttonTextKey: 'OrderDetailsPage.messages.buttonText',
            titleKey: 'OrderDetailsPage.messages.title',
            dateValueKey: 'OrderDetailsPage.messages.dateValue',
            senderValueKey: 'OrderDetailsPage.messages.senderValue',
        },
        orderActionsSectionCardConfig: {
            type: 'orderActions',
            titleKey: 'OrderDetailsPage.orderActions.title',
            completionTemplateButtonXVM: {
                variant: 'fill',
                secondary: true,
                textKey: 'OrderDetailsPage.orderActions.completion.text',
                link: {
                    href: 'assets/lepke.jpg',
                },
            },
            closeOrderButtonXVM: {
                textKey: 'OrderDetailsPage.orderActions.close.text',
                secondary: true,
                variant: 'fill',
            },
        },
        orderDetailsSectionCardConfig: {
            type: 'orderDetails',
            titleKey: 'OrderDetailsPage.details.title',
            infoPanelConfig: {
                clientInfoPanel: {
                    titleKey: 'OrderDetailsPage.details.client.title',
                    addressLabelKey: 'OrderDetailsPage.details.addressLabel',
                    contactLabelKey: 'OrderDetailsPage.details.contactLabel',
                    emailLabelKey: 'OrderDetailsPage.details.emailLabel',
                    phoneLabelKey: 'OrderDetailsPage.details.phoneLabel',
                },
                summaryInfoPanel: {
                    averageDoseLabelKey:
                        'OrderDetailsPage.details.averageDoseLabel',
                    averageDoseValueKey:
                        'OrderDetailsPage.details.averageDoseValue',
                    orderValueLabelKey:
                        'OrderDetailsPage.details.orderValueLabel',
                    orderValueValueKey:
                        'OrderDetailsPage.details.orderValueValue',
                    titleKey: 'OrderDetailsPage.details.summary.title',
                    totalSupplyLabelKey:
                        'OrderDetailsPage.details.totalSupplyLabel',
                    totalSupplyValueKey:
                        'OrderDetailsPage.details.totalSupplyValue',
                    treatmentLabelKey:
                        'OrderDetailsPage.details.treatmentLabel',
                },
            },
        },
    },
};

export const enMock = {
    idTitle: 'id',
    clientText: 'client',
    createdDateText: 'createdDate',
    createdDateValue: 'createdDateV',
    totalAreaText: 'totalArea',
    totalAreaValue: 'totalAreaV',
    addNewMissionButtonText: 'addNewMissionButton',

    orderActions: {
        title: 'tit',
        completionButtonText: 'Proof of completion template',
        closeButtonText: 'Close order',
    },
    orderDetails: {
        title: 'title',
        addressLabel: 'address',
        contactLabel: 'contact',
        emailLabel: 'email',
        phoneLabel: 'phone',
        averageDoseLabel: 'dose',
        averageDoseValue: 'avg',
        orderValueLabel: 'orderL',
        orderValueValue: 'orderV',
        totalSupplyLabel: 'supplyL',
        totalSupplyValue: 'supplyV',
        treatmentLabel: 'treatment',
    },
    messages: {
        title: 'title',
        dateValue: 'date',
        buttonText: 'button',
        senderValue: 'send',
    },
};

export const orderDetailsPageMockConfig: OrderDetailsPageConfig = {
    confirmationDialogVM: {
        closeButtonXVM: {
            icon: MatIcon.ADD,
            secondary: false,
            variant: 'ghost',
        },
        titleKey: '',
        confirmTextKey: '',
        confirmButtonXVM: {
            secondary: false,
            variant: 'fill',
            textKey: enMock.orderActions.completionButtonText,
        },
        cancelButtonXVM: {
            secondary: false,
            variant: 'fill',
            textKey: enMock.orderActions.closeButtonText,
        },
    },
    headerConfig: {
        idTitleKey: enMock.idTitle,
        clientTextKey: enMock.clientText,
        createdDateTextKey: enMock.createdDateText,
        createdDateValueKey: enMock.createdDateValue,
        totalAreaTextKey: enMock.totalAreaText,
        totalAreaValueKey: enMock.totalAreaValue,
        addMissionButtonXVM: {
            secondary: false,
            variant: 'fill',
            textKey: enMock.addNewMissionButtonText,
        },
    },
    sectionCardConfigs: {
        messagesSectionCardConfig: {
            buttonTextKey: enMock.messages.buttonText,
            titleKey: enMock.messages.title,
            dateValueKey: enMock.messages.dateValue,
            senderValueKey: enMock.messages.senderValue,
        },
        orderActionsSectionCardConfig: {
            type: 'orderActions',
            titleKey: enMock.orderActions.title,
            completionTemplateButtonXVM: {
                variant: 'fill',
                secondary: true,
                textKey: enMock.orderActions.completionButtonText,
                link: {
                    href: 'assets/lepke.jpg',
                },
            },
            closeOrderButtonXVM: {
                textKey: enMock.orderActions.closeButtonText,

                secondary: true,
                variant: 'ghost',
            },
        },
        orderDetailsSectionCardConfig: {
            type: 'orderDetails',
            titleKey: enMock.orderDetails.title,
            infoPanelConfig: {
                clientInfoPanel: {
                    titleKey: enMock.orderDetails.title,
                    addressLabelKey: enMock.orderDetails.addressLabel,
                    contactLabelKey: enMock.orderDetails.contactLabel,
                    emailLabelKey: enMock.orderDetails.emailLabel,
                    phoneLabelKey: enMock.orderDetails.phoneLabel,
                },
                summaryInfoPanel: {
                    averageDoseLabelKey: enMock.orderDetails.averageDoseLabel,
                    averageDoseValueKey: enMock.orderDetails.averageDoseValue,
                    orderValueLabelKey: enMock.orderDetails.orderValueLabel,
                    orderValueValueKey: enMock.orderDetails.orderValueValue,
                    titleKey: enMock.orderDetails.title,
                    totalSupplyLabelKey: enMock.orderDetails.totalSupplyLabel,
                    totalSupplyValueKey: enMock.orderDetails.totalSupplyValue,
                    treatmentLabelKey: enMock.orderDetails.treatmentLabel,
                },
            },
        },
    },
};
