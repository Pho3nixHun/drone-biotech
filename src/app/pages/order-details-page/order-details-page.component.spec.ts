import { ComponentFixture, TestBed } from '@angular/core/testing';
import { OrderDetailsPageComponent } from './order-details-page.component';
import { ORDER_DETAILS_PAGE_CONFIG } from './order-details-page.config';
import { enMock, orderDetailsPageMockConfig } from './order-details-page.mock';
import { getTranslocoModule } from 'transloco-testing.module';
import { provideMockAuthStore } from '@stores/auth/auth.testing';

describe('OrderDetailsPageComponent', () => {
    let fixture: ComponentFixture<OrderDetailsPageComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [
                OrderDetailsPageComponent,
                getTranslocoModule({
                    langs: { en: enMock },
                    translocoConfig: {
                        availableLangs: ['en'],
                        defaultLang: 'en',
                    },
                }),
            ],
            providers: [
                {
                    provide: ORDER_DETAILS_PAGE_CONFIG,
                    useValue: orderDetailsPageMockConfig,
                },
                provideMockAuthStore(),
            ],
        }).compileComponents();

        fixture = TestBed.createComponent(OrderDetailsPageComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should render the template correctly', () => {
        // Arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
