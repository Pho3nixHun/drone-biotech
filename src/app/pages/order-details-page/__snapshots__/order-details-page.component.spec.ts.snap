// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`OrderDetailsPageComponent should render the template correctly 1`] = `
<div
  id="root0"
  ng-version="20.1.7"
>
  <!--container-->
  <app-confirmation-dialog>
    <dialog
      class="my-dialog"
    >
      <app-dialog-layout
        class="w-80 md:w-96 block"
      >
        <section
          class="flex flex-col p-4 md:p-10 shadow-md bg-white space-y-4"
        >
          <div
            class="flex flex-row justify-between"
          >
            <h2>
                
            </h2>
            <app-button
              close=""
            >
              <button
                class="btn flex flex-row justify-center btn-primary btn-text"
              >
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  add
                </mat-icon>
                <!--container-->
              </button>
            </app-button>
          </div>
          <div
            class="space-y-4"
          >
            <p>
              
            </p>
          </div>
          <div
            class="ml-auto join"
          >
            <app-button>
              <button
                class="btn flex flex-row justify-center btn-primary"
              >
                 Proof of completion template 
                <!--container-->
              </button>
            </app-button>
            <app-button>
              <button
                class="btn flex flex-row justify-center btn-primary"
              >
                 Close order 
                <!--container-->
              </button>
            </app-button>
          </div>
        </section>
      </app-dialog-layout>
    </dialog>
    <!--container-->
  </app-confirmation-dialog>
  <!--container-->
</div>
`;
